Source/creature_internal.cpp

```diff
@@ -8,9 +8,10 @@ creature_internal::creature_internal(creature_metadata* creature_metadata, stoch
     _creature_metadata = creature_metadata;
     _stochastic_instance = stochastic_instance;
 
+    genetic_factory& factory = genetic_factory::get_instance();
     for(size_t i = 0; i < _creature_metadata->num_chromosomes; i++) {
         chromosome_metadata* chromosome_metadata = _creature_metadata->chromosome_metadata[i];
-        _chromosomes[chromosome_metadata->name] = genetic_factory::create_chromosome(chromosome_metadata);
+        _chromosomes[chromosome_metadata->name] = factory.create_chromosome(chromosome_metadata);
     }
```

Comment:
Why can I use a `.` instead of a `::`?

Reply:
This gets into the difference between a function declared as static vs one that was not. The static keyword makes the function work for the whole class rather than a particular instance. This means static classes can't access state variables because the data contained in them would very depending on what instance you are looking at.  Static methods are accessed with the :: operator. Stylistically I don't like to make purely static classes that need state variables. Since state was added to the factory in the form of registered 'try_create' constructors I switched it from being purely static to a singleton. Thus the method access operator changed from :: to .

Reply:
Is there a significant difference between being a singleton and a static-member?

Reply:
Yes in terms of memory layout but semantically they are very similar.